src/fingerprint_manager.py

```
@@ -1,38 +1,99 @@
+        viewports: Sequence[Dict[str, int]] | None = None,
+        rand: random.Random | None = None,
+    ) -> None:
+        if user_agent_manager is None:
             raise ValueError("Se debe proporcionar un UserAgentManager.")
-        if not viewports:
+        self.user_agent_manager = user_agent_manager
+
+        # Use a copy of the default list to avoid accidental mutation.
+        self.viewports: list[Dict[str, int]] = list(viewports) if viewports else list(DEFAULT_VIEWPORTS)
+        if not self.viewports:
             raise ValueError("La lista de viewports no puede estar vacía.")
```

Ruff (0.12.2)

Avoid specifying long messages outside the exception class

(TRY003)